File: frontend/src/features/file-search/schemas.ts
import { z } from 'zod'

export const fileSearchDocumentSchema = z.object({
  id: z.string(),
  chatSessionId: z.string(),
  originalName: z.string(),
  contentType: z.string(),
  sizeBytes: z.number(),
  status: z.enum(['pending', 'indexing', 'ready', 'failed']),
  failureMessage: z.string().nullable(),
  createdAt: z.string(),
  updatedAt: z.string(),
  readyAt: z.string().nullable(),
})

export const fileSearchDocumentResponseSchema = z.object({
  success: z.boolean(),
  data: fileSearchDocumentSchema.nullable(),
})

export const fileSearchCitationSchema = z.object({
  id: z.string(),
  documentId: z.string(),
  position: z.number(),
  fileName: z.string(),
  pageNumber: z.number().nullable(),
  sourceExcerpt: z.string(),
  mediaId: z.string().nullable(),
  customMetadata: z.record(z.string(), z.unknown()).nullable(),
})

export const fileSearchDeleteResponseSchema = z.object({
  success: z.boolean(),
  message: z.string(),
})

export type FileSearchDocument = z.infer<typeof fileSearchDocumentSchema>
export type FileSearchCitation = z.infer<typeof fileSearchCitationSchema>

File: backend/app/file_search/schemas.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field

DocumentStatus = Literal["pending", "indexing", "ready", "failed"]


class FileSearchDocumentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: str
    chat_session_id: str = Field(serialization_alias="chatSessionId")
    original_name: str = Field(serialization_alias="originalName")
    content_type: str = Field(serialization_alias="contentType")
    size_bytes: int = Field(serialization_alias="sizeBytes")
    status: DocumentStatus
    failure_message: str | None = Field(default=None, serialization_alias="failureMessage")
    created_at: datetime = Field(serialization_alias="createdAt")
    updated_at: datetime = Field(serialization_alias="updatedAt")
    ready_at: datetime | None = Field(default=None, serialization_alias="readyAt")


class FileSearchDocumentResponse(BaseModel):
    success: bool = True
    data: FileSearchDocumentRead | None


class FileSearchCitationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: str
    document_id: str = Field(serialization_alias="documentId")
    position: int
    file_name: str = Field(serialization_alias="fileName")
    page_number: int | None = Field(default=None, serialization_alias="pageNumber")
    source_excerpt: str = Field(serialization_alias="sourceExcerpt")
    media_id: str | None = Field(default=None, serialization_alias="mediaId")
    custom_metadata: dict[str, Any] | None = Field(
        default=None,
        serialization_alias="customMetadata",
    )


class FileSearchDeleteResponse(BaseModel):
    success: bool = True
    message: str

